opentech/apply/funds/views.py

```diff
@@ -466,23 +466,77 @@ class ReviewerSubmissionDetailView(ReviewContextMixin, ActivityContextMixin, Del
     model = ApplicationSubmission
     form_views = [CommentFormView]
 
+    def dispatch(self, request, *args, **kwargs):
+        submission = self.get_object()
+        # If the requesting user submitted the application, return the Applicant view.
+        # Reviewers and partners may somtimes be appliants as well.
+        if submission.user == request.user:
+            return ApplicantSubmissionDetailView.as_view()(request, *args, **kwargs)
+        # Only allow reviewers in the submission they are added as reviewers or has reviewed earlier
+        reviewer_has_access = submission.reviewers.filter(pk=request.user.pk).exists() or submission.reviews.values('author').filter(author=request.user.pk)
```

Comment:
New take on this from OTF. Reviewers should be able to view any submission that have reached an "AC review" state or higher and have not been dismissed.

Can we make use of "permissions.can_view()" for this? Or should we add something similar to COMMUNITY_REVIEW_PHASES where we list all the states reviewers should have access to, i.e. REVIEW_VIEW_ACCESS_PHASES? Other better solutions?

(Reviewers should, as it is now and have been for a long time, only be able to review submission they have been assigned to.)

Reply:
If this needs some thinking we can revert to the current functionality where reviewers can see all submission. It has been like that from day one so no harm in postponing updating this a bit.

Reply:
Getting the other nice improvements in this PR out is more important.

Reply:
Thanks for the information. We can use permissions.can_view() along with other required checks. It seems better to handle it in another PR. We also need to change `ReviewerDeterminationDetailView` as currently, a reviewer can see any determination as well.